api/server/sdk/volume_node_ops.go

```diff
@@ -192,13 +192,14 @@ func (s *VolumeServer) Mount(
 			attachOptions[mountattachoptions.OptionsSecretKey] = req.Options.SecretKey
 			attachOptions[mountattachoptions.OptionsSecretContext] = req.Options.SecretContext
 		}
+
 		if vol.Scaled() {
 			vol, err = s.attachScale(ctx, vol, attachOptions)
 		} else {
 			vol, err = s.attachVol(ctx, vol, attachOptions)
 		}
 		if err != nil {
-			return nil, status.Error(codes.InvalidArgument, err.Error())
+			return nil, err
```

Comment:
Here's a status of which codes we implement:

**CreateVolume**

| Code | Implemented | Description |
| - | - | - |
| INVALID_ARGUMENT | Yes | |
| NOT_FOUND  | Yes | | 
| ALREADY_EXISTS | Yes | |
| RESOURCE_EXHAUSTED  | No | Indicates that although the accessible_topology field is valid, a new volume can not be provisioned with the specified topology constraints. More human-readable information MAY be provided in the gRPC status.message field.|
| OUT_OF_RANGE | No | Indicates that the capacity range is not allowed by the Plugin, for example when trying to create a volume smaller than the source snapshot. More human-readable information MAY be provided in the gRPC status.message field. |

**DeleteVolume:** 

| Code | Implemented | Description |
| - | - | - |
| FAILED_PRECONDITION | No | Indicates that the volume corresponding to the specified volume_id could not be deleted because it is in use by another resource. |

**CreateSnapshot**

| Code | Implemented | Description |
| - | - | - |
| ALREADY_EXISTS | No | Indicates that a snapshot corresponding to the specified snapshot name already exists but is incompatible with the specified volume_id. |
| ABORTED | No | Indicates that there is already an operation pending for the specified snapshot. In general the Cluster Orchestrator (CO) is responsible for ensuring that there is no more than one call "in-flight" per snapshot at a given time. However, in some circumstances, the CO MAY lose state (for example when the CO crashes and restarts), and MAY issue multiple calls simultaneously for the same snapshot. The Plugin, SHOULD handle this as gracefully as possible, and MAY return this error code to reject secondary calls. |
| RESOURCE_EXHAUSTED | No | There is not enough space on the storage system to handle the create snapshot request. |

**DeleteSnapshot**

| Code | Implemented | Description |
| - | - | - |
| FAILED_PRECONDITION | No | Indicates that the snapshot corresponding to the specified snapshot_id could not be deleted because it is in use by another resource. |
| ABORTED | No | Indicates that there is already an operation pending for the specified snapshot. In general the Cluster Orchestrator (CO) is responsible for ensuring that there is no more than one call "in-flight" per snapshot at a given time. However, in some circumstances, the CO MAY lose state (for example when the CO crashes and restarts), and MAY issue multiple calls simultaneously for the same snapshot. The Plugin, SHOULD handle this as gracefully as possible, and MAY return this error code to reject secondary calls. |

**NodePublishVolume**

| Code | Implemented | Description |
| - | - | - |
| NOT_FOUND | Yes | |
| ALREADY_EXISTS | No | Indicates that a volume corresponding to the specified volume_id has already been published at the specified target_path but is incompatible with the specified volume_capability or readonly flag. |
| FAILED_PRECONDITION | No | Indicates that the CO has exceeded the volume's capabilities because the volume does not have MULTI_NODE capability. |


**NodeUnpublishVolume**

| Code | Implemented | Description |
| - | - | - |
| NOT_FOUND | Yes | |

Reply:
Ignore the above ^ was checking all of the error codes to see if we implemented them.

We're good with the error codes we are using - they follow the spec description.